src/test/java/com/microsoft/sqlserver/testframework/DBResultSet.java

```diff
@@ -85,4 +110,327 @@ public Object getObject(int index) throws SQLException {
     public void updateObject(int index) throws SQLException {
         // TODO: update object based on cursor type
     }
+
+    /**
+     * 
+     * @throws SQLException
+     */
+    public void verify(DBTable table) throws SQLException {
+        currentTable = table;
+        metaData = this.getMetaData();
+        metaData.verify();
+
+        while (this.next())
+            this.verifyCurrentRow(table);
+    }
+
+    /**
+     * @throws SQLException
+     * 
+     */
+    public void verifyCurrentRow(DBTable table) throws SQLException {
+        currentTable = table;
+        int totalColumns = ((ResultSet) product()).getMetaData().getColumnCount();
+
+        Class _class = Object.class;
+        for (int i = 0; i < totalColumns; i++)
+            verifydata(i, _class, null);
+
+    }
+
+    /**
+     * 
+     * @param ordinal
+     * @param coercion
+     * @param arg
+     * @throws SQLException
+     */
+    public void verifydata(int ordinal, Class coercion, Object arg) throws SQLException {
+        Object backendData = this.currentrow().get(ordinal);
```

Comment:
The name backendData gives an impression that it is actual data from DB, when it is actually expected data. It would be better to rename this variable.